storage/qemu_spdk_setup.md

```diff
@@ -0,0 +1,146 @@
+
+# OPI QEMU SPDK setup
+
+## Docs
+
+From <https://spdk.io/doc/vhost.html>
+
+## Virtualization support
+
+Make sure that VT-x/AMD-v support is enabled in BIOS
+
+```bash
+$ lscpu | grep -i virtualization
+Virtualization:                  VT-x
+```
+
+and that kvm modules are loaded
+
+```bash
+$ lsmod | grep -i kvm
+kvm_intel             217088  0
+kvm                   614400  1 kvm_intel
+irqbypass              16384  1 kvm
+```
+
+## Tool installation
+
+### wget
+
+Installation on Fedora
+
+```bash
+sudo dnf install wget
+```
+
+or on Ubuntu
+
+```bash
+sudo apt install wget
+```
+
+### qemu
+
+```bash
+sudo dnf install qemu-kvm
+```
+
+or
+
+```bash
+sudo apt install qemu-system
+```
+
+### libguestfs-tools
+
+```bash
+sudo dnf install libguestfs-tools-c
+```
+
+or
+
+```bash
+sudo apt install libguestfs-tools
+```
+
+## Huge Pages
+
+```bash
+echo 4096 | sudo tee /proc/sys/vm/nr_hugepages
+```
+
+## Run SPDK
+
+```bash
+./spdk/build/bin/spdk_tgt  -S /var/tmp -s 1024 -m 0x3
```

Comment:
What is the context here?  Is this directly on the host?  Where did spdk_tgt build come from?

Reply:
This page explores further emulation of our environment with Host and xPU in it

- QEMU is emulating our Host now, in this context
- SPDK still runs on xPU (skipped container for now) but the build comes from docker, for example
- SPDK implements Virtio target (in this example virtio-blk, but also virtio-scsi or virtio-fs are possible to expand on later)
- VM runs virtio drivers and see block devices and can run I/O to/from them which goes through SPDK to the remote storage